### PR TITLE
feat(vrl): add `parse_groks` function with multiple grok patterns and aliases

### LINKED ISSUE #9660: Support inline aliases (helper rules) in the `parse_grok` function
It's common for Grok parsers to allow for aliases, or helper rules, for patterns that simplify Grok parsing rules. Because VRL already have a `patterns` argument, I propose that we add an optional `aliases` argument that can be used in the patterns:

```
. |= parse_grok(
    .message,
    patterns: [
        "%{_s3_bucket_owner} %{_s3_bucket} %{_date_access} (?>%{_client_ip}|-) %{_client_id} %{_request_id} %{_s3_operation} %{notSpace} "(?>%{_method} |)%{_url}(?> %{_version}|)" %{_status_code} %{_s3_error_code} (?>%{_bytes_written}|-) (?>%{_object_size}|-) %{_duration} (?>%{_request_processing_time}|-) "%{_referer}" "%{_user_agent}" %{_request_version_id}.*",
        "%{_s3_bucket_owner} %{_s3_bucket} %{_date_access} (?>%{_client_ip}|-) %{_client_id} %{_request_id} %{_s3_operation}.*"
    ]
    aliases: {
        "_s3_bucket_owner": "%{notSpace:s3.bucket_owner}",
        "_s3_bucket": "%{notSpace:s3.bucket}",
        "_s3_operation": "%{notSpace:s3.operation}",
        "_s3_error_code": "%{notSpace:s3.error_code:nullIf("-")}",
        "_request_processing_time": "%{integer:http.request_processing_time}",
        "_request_id": "%{notSpace:http.request_id}",
        "_request_version_id": "%{notSpace:http.request_version_id:nullIf("-")}",
        "_bytes_written": "%{integer:network.bytes_written}",
        "_bytes_read": "%{integer:network.bytes_read}",
        "_object_size": "%{integer:network.object_size}",
        "_client_ip": "%{ipOrHost:network.client.ip}",
        "_client_id": "%{notSpace:network.client.id}",
        "_version": "HTTP\/%{regex(\"\\d+\\.\\d+\"):http.version}",
        "_url": "%{notSpace:http.url}",
        "_ident": "%{notSpace:http.ident:nullIf("-")}",
        "_user_agent": "%{regex(\"[^\\\\"]*"):http.useragent}",
        "_referer": "%{notSpace:http.referer:nullIf("-")}",
        "_status_code": "%{integer:http.status_code}",
        "_method": "%{word:http.method}",
        "_duration": "%{integer:duration:scale(1000000)}",
        "_date_access \[%{date(\"dd/MMM/yyyy:HH:mm:ss Z\"):date_access}\]"
    }
)
```

### PR DESCRIPTION
**Pull request by vladimir-dd**
Introduces a new VRL function `parse_groks`, which supports multiple grok patterns and aliases.

```
parse_groks!(
	"2020-10-02T23:22:12.223222Z info Hello world",
	patterns: [
		"%{common_prefix} %{_status} %{_message}",
		"%{common_prefix} %{_message}",
	],
	aliases: {
		common_prefix: "%{_timestamp} %{_loglevel}",
		_timestamp: "%{TIMESTAMP_ISO8601:timestamp}",
		_loglevel: "%{LOGLEVEL:level}",
		_status: "%{POSINT:status}",
		_message: "%{GREEDYDATA:message}"
	}
)
```

Closes #9660, #9661

### CODE REVIEW

**Comment by StephenWakely** [comment]
I think this could be a bit confusing. `parse_grok` uses the grok library while `parse_groks` uses the Datadog grok library.

I'm not sure what the real difference between these two libraries are. If there is no actual difference in functionality, maybe we should move `parse_grok` to use the Datadog grok library, but if there is a difference I think we should rename this to be `parse_datadog_groks` or similar.

**Comment by jszwedko** [comment]
> I think this could be a bit confusing. `parse_grok` uses the grok library while `parse_groks` uses the Datadog grok library.
> 
> I'm not sure what the real difference between these two libraries are. If there is no actual difference in functionality, maybe we should move `parse_grok` to use the Datadog grok library, but if there is a difference I think we should rename this to be `parse_datadog_groks` or similar.

Yeah, agreed. We discussed this a bit today. If the Datadog grok parsing library is a strict superset and there are no downsides (like performance) than I'm inclined to just have one set of `parse_grok` and `parse_groks` that uses the internal Datadog grok parsing library. We would just update the docs to showcase the additional functionality it supports on top of standard grok parsing.

**Comment by vladimir-dd** [comment]
Yes, DD grok is a strict superset of normal grok expression with extra filters and matchers.

File: lib/vrl/stdlib/src/parse_datadog_grok.rs [hunk 1]
```diff
@@ -0,0 +1,326 @@
+use datadog_grok::{
```
Comment (StephenWakely) [inline lib/vrl/stdlib/src/parse_datadog_grok.rs:1]: From what I can see, this file isn't being used? Can we remove it?

File: website/cue/reference/remap/functions/parse_groks.cue [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+		},
+		{
+			name:        "patterns"
+			description: "The [Grok patterns](https://github.com/daschl/grok/tree/master/patterns), which are tried in order until the first match."
+			required:    true
+			type: ["array"]
+		},
+		{
+			name:        "remove_empty"
+			description: "If set to `true`, any patterns that resolve to an empty value are removed from the result."
```
Comment (StephenWakely) [inline website/cue/reference/remap/functions/parse_groks.cue:31]: I'm not sure I understand. How does a pattern resolve to an empty value?

Comment (StephenWakely) [reply website/cue/reference/remap/functions/parse_groks.cue:31]: Could it say?

```suggestion
			description: "If set to `true`, any patterns that resolve to an empty string are removed from the result."
```

**Review by StephenWakely — APPROVED** [review]
Yes! Looking good. I think that extra file can be deleted?

We need to make sure that soon we can take this approach and apply it to `parse_grok` so that we are using just one grok library throughout VRL.

File: lib/vrl/stdlib/src/parse_datadog_grok.rs (hunk 1 above)
Comment (vladimir-dd) [reply lib/vrl/stdlib/src/parse_datadog_grok.rs:1]: Oh, right, indeed, thanks 👍

**Pull request merged**